test/indexing.jl

```diff
@@ -8,34 +8,44 @@ using Test, DataFrames
     @test df[1] == [1, 2, 3]
     @test df[1] === eachcol(df)[1]
     @test df[1:2] == DataFrame(a=1:3, b=4:6)
+    @test df[r"[ab]"] == DataFrame(a=1:3, b=4:6)
     @test eachcol(df, false)[1] === df[1]
     @test eachcol(view(df,1:2), false)[1] == eachcol(df, false)[1]
     @test eachcol(df[1:2], false)[1] == eachcol(df, false)[1]
+    @test eachcol(df[r"[ab]"], false)[1] == eachcol(df, false)[1]
     @test eachcol(df[1:2], false)[1] !== eachcol(df, false)[1]
     @test df[:] == df
+    @test df[r""] == df
     @test df[:] !== df
```

Comment:
Also test this for regexes? Same for other things in this test set.

Reply:
fixed